io_mesh_w3d/common/structs/mesh_structs/shader_material.py

```diff
@@ -39,7 +38,13 @@ def write(self, io_stream):
 class ShaderMaterialProperty(Struct):
     type = 0
     name = ''
-    value = None
+    value = Vector((1.0, 1.0, 1.0, 1.0))
+
+    def to_rgb(self):
+        return self.value.x, self.value.y, self.value.z
+
+    def to_rgba(self):
+        return self.value.x, self.value.y, self.value.z, self.value.w if len(self.value) > 3 else 1.0
```

Comment:
check could be avoided if self.value.w defaulted to 1.0 (which it does)

Reply:
but sometimes we read only a vec 3, especially in w3d file format, then it fails with an index out of range